cypress/utils/commands.js

```diff
@@ -30,10 +30,49 @@ Cypress.Commands.add('getElementsByTestIds', (testIds, options = {}) => {
   return cy.get(selectors.join(','), options);
 });
 
-Cypress.Commands.add('getElementByDataTestId', (testId) => {
-  return cy.get(`[data-testid="${testId}"]`);
+/**
+ * Get DOM elements with a data-test-subj id containing the testId.
+ * @param {string} testId data-test-subj value.
+ * @param {object} options get options. Default: {}
+ * @example
+ * // returns all DOM elements that has a data-test-subj including the string 'table'
+ * cy.getElementsByTestIdLike('table')
+ */
+Cypress.Commands.add('getElementsByTestIdLike', (partialTestId, options = {}) => {
+  return cy.get(`[data-test-subj*="${partialTestId}"]`, options);
 });
 
+/**
+ * Find element from previous chained element with a data-test-subj id containing the testId.
+ * @param {string} subject DOM object to find within.
+ * @param {string} testId data-test-subj value.
+ * @param {object} options get options. Default: {}
+ * @example
+ * // returns all DOM elements that has a data-test-subj including the string 'table'
+ * cy.findElementsByTestIdLike('table')
+ */
+Cypress.Commands.add(
+  'findElementByTestIdLike',
+  { prevSubject: true },
+  (subject, partialTestId, options = {}) => {
+    return cy.wrap(subject).find(`[data-test-subj*="${partialTestId}"]`, options);
+  }
+);
+
+/**
+ * Find element from previous chained element by data-test-subj id.
+ * @param {string} subject DOM object to find within.
+ * @param {string} testId data-test-subj value.
+ * @param {object} options get options. Default: {}
+ */
+Cypress.Commands.add(
+  'findElementByTestId',
+  { prevSubject: true },
+  (subject, testId, options = {}) => {
+    return cy.wrap(subject).find(`[data-test-subj="${testId}"]`, options);
+  }
+);
```

Comment:
Could we update the associated `index.d.ts` to include typings for these?

Reply:
I time boxed a couple hours trying to switch to ts with no avail. I'll try again on Monday given you've successfully added typing.

Reply:
Cypress uses the term **get** for the entire DOM and **find** for within the previously selected element. I think sticking with that convention is better.

Reply:
Added typings.